sdks/go/pkg/beam/core/runtime/harness/worker_status.go

```diff
@@ -65,20 +70,50 @@ func (w *workerStatusHandler) start(ctx context.Context) error {
 	return nil
 }
 
+func memoryUsage() string {
+	m := runtime.MemStats{}
+	runtime.ReadMemStats(&m)
+	return fmt.Sprintf("\n Total Alloc: %v bytes \n Sys: %v bytes \n Mallocs: %v\n Frees: %v\n HeapAlloc: %v bytes", m.TotalAlloc, m.Sys, m.Mallocs, m.Frees, m.HeapAlloc)
```

Comment:
Aside from the previous string builder, we get *so much* information in MemStats, we should at least try to match what Java's putting out.

https://pkg.go.dev/runtime#MemStats  -> Everything we get.
Java's print out: https://github.com/apache/beam/blob/ffef8de04a93435e69faf3bf65efe11852cbd8dc/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/status/MemoryMonitor.java#L622 

We can do the same level of readability improvement that Java is doing, though I don't think we can do "thrashing". We can show the GC target size, and `GCCPUFraction` gives us how much CPU time has been spent in GC in a way we can use as the percentage. Not sure about "thrashing" and "pushback", but this gets us close to the way there.

Java's for reference:
https://github.com/apache/beam/blob/ffef8de04a93435e69faf3bf65efe11852cbd8dc/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/status/MemoryMonitor.java#L411

`NextGC` is useful since it sets the target of the next GC.

`Mallocs` & `Frees` aren't super useful outside of a full heap dump (we care about memory, not counts). `HeapInuse` is more useful, as is `StackInuse` and `StackSys` (We'll need both for a complete picture, as Go tries to do as much as possible on the stack, instead of the heap, unlike Java which is All Heap).